src/apppermission.cpp

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "apppermission.h"
+#include "logger.h"
+#include "mozillavpn.h"
+#include "settingsholder.h"
+#include "applistprovider.h"
+#include <QVector>
+
+// TODO :: #ifdef
+#include "platforms/dummy/dummyapplistprovider.h"
+// #endif
+
+
+namespace {
+Logger logger(LOG_MAIN, "AppPermission");
+AppPermission* s_instance = nullptr;
+}
+
+AppPermission::AppPermission()
+{
+ //   MVPN_COUNT_CTOR(AppPermission);
+    Q_ASSERT(!s_instance);
+    s_instance = this;
+    // Todo: ifdef for plattforms
+    m_listprovider = new DummyAppListProvider();
+    // #endif
+    connect(m_listprovider, &AppListProvider::newAppList, this,
+            &AppPermission::reciveAppList);
+
+    m_listprovider->getApplicationList();
```

Comment:
we should retrieve the list only when the view is visible. Otherwise we do not refresh the list when needed.